### PR TITLE
[deckhouse-controller] Fixed tag update

### PR DESCRIPTION
**Pull request by libmonsoon-dev**
## Description
Fixed artifacts remaining from previous versions where snapshots were used

## Why do we need it, and what problem does it solve?
Deckhouse controller could fail with panic if we have some Pending/Terminating deckhouse pod

## Why do we need it in the patch release (if we do)?
Deckhouse controller could fail with panic if we have some Pending/Terminating deckhouse pod

## Checklist
- [x] The code is covered by unit tests.
- [ ] e2e tests passed.
- [ ] Documentation updated according to the changes.
- [ ] Changes were tested in the Kubernetes cluster manually.

## Changelog entries

```changes
section: deckhouse-controller
type: fix
summary: Fixed panic when processing terminating deckhouse pod
impact_level: default
```

### CODE REVIEW

File: deckhouse-controller/pkg/controller/deckhouse-release/controller.go [hunk 1]
```diff
@@ -551,6 +557,18 @@ func (r *deckhouseReleaseReconciler) tagUpdate(ctx context.Context, pods []corev
 	return nil
 }
 
+func getNewestPod(pods []corev1.Pod) corev1.Pod {
+	result := pods[0]
+
+	for _, pod := range pods[1:] {
```
Comment (yalosev) [inline deckhouse-controller/pkg/controller/deckhouse-release/controller.go:563]: I guess we can filter a pod [here](https://github.com/deckhouse/deckhouse/pull/9621/files#diff-f5f1549dccbd7351a2389d6bc059b5f448e7e169efc48647e51bb86c2a1ec718R398). No need to return the full list and then filter it for the second time

Comment (miklezzzz) [reply deckhouse-controller/pkg/controller/deckhouse-release/controller.go:563]: Why not return a full list of deckhouse pods? as I see it - we check all pods' statuses and cancel update if any pod isn't ready or in pending state.

**Review by miklezzzz — APPROVED** [review]

File: deckhouse-controller/pkg/controller/deckhouse-release/controller.go (hunk 1 above)
Comment (miklezzzz) [reply deckhouse-controller/pkg/controller/deckhouse-release/controller.go:563]: Why not return a full list of deckhouse pods? as I see it - we check all pods' statuses and cancel update if any pod isn't ready or in pending state.

Comment (yalosev) [reply deckhouse-controller/pkg/controller/deckhouse-release/controller.go:563]: We need to figure out the image hash of the last running deckhouse pod, and decide, do we have to update or not.
I don't see the reason, why do we need the full list of deckhouse pods.

**Review by yalosev — APPROVED** [review]

**Comment by deckhouse-BOaTswain** [comment]
Failure: cherry-pick commit ec968fd4932f6a278828e02f7ee97858f41dabcf to the branch [release-1.63](https://github.com/deckhouse/deckhouse/tree/release-1.63) failed. See [Job](https://github.com/deckhouse/deckhouse/actions/runs/10739073286) for details.

**Comment by deckhouse-BOaTswain** [comment]
Cherry pick PR [9641](https://github.com/deckhouse/deckhouse/pull/9641) to the branch [release-1.64](https://github.com/deckhouse/deckhouse/tree/release-1.64) successful!

**Comment by z9r5** [comment]
Cherry-picked into release-1.63 with #9644

**Pull request merged**